runtime/vm/classloadersearch.c

```diff
@@ -237,3 +241,71 @@ addZipToLoader(J9JavaVM * vm, const char * filename, J9ClassLoader * classLoader
 
 	return rc;
 }
+
+jclass
+getJimModules(J9VMThread *currentThread) {
+	J9JavaVM *vm = currentThread->javaVM;
+	if (NULL == vm->jimModules) {
+		const J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+		/* no need for mutual exclusion since repeating this is harmless */
+		J9Class *j9JimModules = vmFuncs->internalFindKnownClass(currentThread,
+				J9VMCONSTANTPOOL_JDKINTERNALMODULEMODULES,
+				J9_FINDKNOWNCLASS_FLAG_INITIALIZE | J9_FINDKNOWNCLASS_FLAG_NON_FATAL);
+		if (NULL != j9JimModules) {
+			vm->jimModules = (jclass)&j9JimModules->classObject;
```

Comment:
We should add a macro for this later, even though it will look odd to J9CLASS_FROM_JCLASS followed by JCLASS_FROM_J9CLASS.

Reply:
Also probably worth a comment here that the address of j9class->classObject is considered to be a JNI global ref.  It won't keep a class alive, but that's OK here since the class in question is not a candidate for unloading.

Reply:
https://github.com/eclipse/openj9/issues/2437